supernode/daemon/mgr/scheduler/manager.go

```diff
@@ -203,34 +205,40 @@ func (sm *Manager) tryGetPID(ctx context.Context, taskID string, pieceNum int, p
 		// if failed to get peerState, and then it should not be needed.
 		peerState, err := sm.progressMgr.GetPeerStateByPeerID(ctx, peerIDs[i])
 		if err != nil {
+			logrus.Warnf("scheduler: failed to GetPeerStateByPeerID taskID(%s) peerID(%s): %v", taskID, peerIDs[i], err)
 			sm.deletePeerIDByPieceNum(ctx, taskID, pieceNum, peerIDs[i])
 			continue
 		}
 
 		// if the service has been down, and then it should not be needed.
 		if peerState.ServiceDownTime != nil && *(peerState.ServiceDownTime) > 0 {
+			logrus.Warnf("scheduler: the peer(%s) has been offline and will delete it from piece state", peerIDs[i])
 			sm.deletePeerIDByPieceNum(ctx, taskID, pieceNum, peerIDs[i])
 			continue
 		}
 
 		// if service has failed for EliminationLimit times, and then it should not be needed.
-		if peerState.ServiceErrorCount != nil && peerState.ServiceErrorCount.Get() >= config.EliminationLimit {
-			sm.deletePeerIDByPieceNum(ctx, taskID, pieceNum, peerIDs[i])
-			continue
+		if peerState.ServiceErrorCount != nil {
+			serviceErrorCount := peerState.ServiceErrorCount.Get()
+			if serviceErrorCount >= config.EliminationLimit {
+				logrus.Warnf("scheduler: the peer(%s) has been eliminated to for too many errors count(%d) as a peer server", peerIDs[i], serviceErrorCount)
```

Comment:
peer(%s) has been eliminated because of too many errors (%d) to remain a peer server